src/System.Data.SqlClient/tests/ManualTests/ProviderAgnostic/ReaderTest/ReaderTest.cs

```diff
@@ -266,5 +270,371 @@ public static void TestMain()
                 }
             }
         }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async Task TestConcurrentLoadSync()
+        {
+            ConcurrentLoadContext context = new ConcurrentLoadContext(
+                providerFactory: SqlClientFactory.Instance,
+                connectionString: DataTestUtility.TcpConnStr,
+                mode: ConcurrentLoadContext.Mode.Async,
+                warmupSeconds: 1,
+                executionSeconds: 60,
+                threadCount: Environment.ProcessorCount * 4
+            );
+            var (transactionPerSecond, average, stdDeviation) = await context.Run();
+            Assert.InRange(transactionPerSecond, 1, int.MaxValue);
+            Assert.True(average > 0);
+            Assert.True(stdDeviation != 0);
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async Task TestConcurrentLoadAsync()
+        {
+            ConcurrentLoadContext context = new ConcurrentLoadContext(
+                providerFactory: SqlClientFactory.Instance,
+                connectionString: DataTestUtility.TcpConnStr,
+                mode: ConcurrentLoadContext.Mode.Async,
+                warmupSeconds: 1,
+                executionSeconds: 60,
+                threadCount: Environment.ProcessorCount * 4
+            );
+            var (transactionPerSecond, average, stdDeviation) = await context.Run();
+            Assert.InRange(transactionPerSecond, 1, int.MaxValue);
+            Assert.True(average > 0);
+            Assert.True(stdDeviation != 0);
+        }
+
+        public class ConcurrentLoadContext
```

Comment:
This class can be moved to another file altogether.

Reply:
done.